src/components/HoursModal/HoursModal.tsx

```diff
@@ -0,0 +1,283 @@
+        : false
+    );
+  };
+
+  //check if the reservation is not overlapping another one
+  const isValidReservationTime = (startTime: Date, endTime: Date) => {
+    const disabledHoursArr = disabledHours();
+    let startHour = 0;
+
+    startHour = Number(format(startTime.getTime(), 'k'));
+    const endHour = Number(format(endTime.getTime(), 'k'));
+
+    if (
+      disabledHoursArr.every(
+        (hour: number) => !(hour > startHour && hour < endHour)
+      )
+    ) {
+      if (startHour == endHour) {
+        setErrorMsg('You can not reserve a desk for less than 1 hour!');
+      } else {
+        setErrorMsg('');
+      }
+    } else {
+      setErrorMsg(
+        'You cannot set this time as end hour, because the desk is already reserved for some hours between!'
+      );
+    }
+  };
+
+  //set values to timepickers
+  const timepickerValueFrom = (value: moment.Moment) => {
+    if (value) {
+      setTimeValueFrom(value.toDate());
+      if (timeValueTo && typeof timeValueTo !== 'number') {
+        isValidReservationTime(value.toDate(), timeValueTo);
+      }
+    }
+  };
+  const timepickerValueTo = (value: moment.Moment) => {
+    if (value) {
+      setTimeValueTo(value.toDate());
+
+      timeValueFrom && value ? setErrorMsg('') : null;
+      if (timeValueFrom && typeof timeValueFrom !== 'number') {
+        isValidReservationTime(timeValueFrom, value.toDate());
+      }
+    } else {
+      setTimeValueTo(0);
+    }
+  };
+
+  //set already reserved hours as disabled to timepickers
+  const disabledHours = () => {
+    const disabledArray: number[] = [
+      0, 1, 2, 3, 4, 5, 6, 7, 8, 19, 20, 21, 22, 23,
+    ];
+
+    if (reservedHours) {
+      const hoursArray = reservedHours.split(',');
+      hoursArray.forEach((el) => {
+        for (
+          let i = Number(format(new Date(el.split('-')[0]).getTime(), 'k'));
+          i <= Number(format(new Date(el.split('-')[1]).getTime(), 'k'));
+          i++
+        ) {
+          disabledArray.push(i);
+        }
+      });
+    }
+    return disabledArray;
+  };
+
+  const handleSave = async () => {
+    //validations before save
+    if (!isReservationOptionChecked) {
+      setErrorMsg('Please, select time option for your reservation!');
+      return;
+    }
+    if (isTimepickerVisible && !timeValueFrom && !timeValueTo) {
+      setErrorMsg('Please, pick a time for your reservation!');
+      return;
+    }
+    if (isAfter(timeValueFrom, timeValueTo)) {
+      setErrorMsg('End time should be after start time!');
+      return;
+    }
+    if (errorMsg) {
+      return;
+    }
+
+    const allDayData = {
+      date: currDate,
+      isAllDayReservation: !isTimepickerVisible,
+      timeFrom: timeValueFrom,
+      timeTo: timeValueTo,
+      deskId,
+      userId,
+    };
+
+    if (deskId) {
+      const imageMapId = shownImageMap;
+      await saveDateRequest(allDayData, imageMapId);
+      await saveDateRequest(allDayData);
+      allDayData.isAllDayReservation

```

Comment:
Use the ternary operator to set the value and call checkIfReservationIsSaved() only once
example:
const status = allDayData.isAllDayReservation ? "value 1" : "value 2"
check...()